guide/src/ecosystem/async-await.md

```diff
@@ -268,3 +269,191 @@ async fn main() -> pyo3::PyResult<()> {
     pyo3_asyncio::testing::main().await
 }
 ```
+
+
+# Calling async Python functions in Rust code
+
+In the previous chapter, we learnt how to call Python functions in Rust code. Now, we will look at two ways to call Python async functions from Rust code.
+In python, even an async function is treated like a regular function until called. When called, an async python function is treated as a coroutine.(More on this below.)
+We can either pass it either as a coroutine or either as a regular function.
+
+
+## Async Functions
+
+Before we start, let us take a look at the following snippet.
+
+```python
+async def fx():
+    return "hello, world"
+```
+
+Above, we have defined an async function. We will now execute it in rust code.
+
+In python, even an async function is treated like a regular function until called.
+e.g.
+
+```python
+>>> type(fx)
+<class 'function'>
+>>> type(fx())
+<class 'coroutine'>
+```
+
+We will now write rust functions to execute the python code in rust. The `PyAny` binding will be used to call Python functions in Rust code.
+
+Now, let us take a look at how will we share the Python function in our rust code.
+
+```rust
+
+use pyo3::prelude::*;
+
+#[pyclass]
+struct SomeObject {}
+
+#[pymethods]
+impl SomeObject {
+    #[new]
+    fn new() -> Self {
+        Self {}
+    }
+
+    fn start(&self, python_function: &PyAny) {
+        // where python_function is the function added from python
+    }
+}
+
+#[pymodule]
+pub fn some_module(py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    m.add_class::<Server>()?;
+    pyo3_asyncio::try_init(py);
+    Ok(())
+}
```

Comment:
I think that using `#[pyclass]` is too complex for this example, let's use a `#[pyfunction]` instead. It will make it easier for the user to focus on the relevant code.

All the examples below will also need to be changed.

```suggestion
use pyo3::prelude::*;

/// Awaits a Python coroutine from Rust code.
fn await_in_rust(coroutine: &PyAny) {
    // we will fill in this implementation as the section continues...
}

// We need to create a module to contain this function
#[pymodule]
pub fn some_module(py: Python<'_>, m: &PyModule) -> PyResult<()> {
    m.add_function(wrap_pyfunction!(await_in_rust, m))?;
    pyo3_asyncio::try_init(py);
    Ok(())
}
```